nestkernel/node_manager.cpp

```diff
@@ -394,11 +391,19 @@ Node* NodeManager::get_node( index gid, thread t )
   return node;
 }
 
-Node* NodeManager::get_node_indp_thread( index gid )
+Node* NodeManager::get_node_or_proxy( index gid )
 {
   thread t = kernel().vp_manager.vp_to_thread( kernel().vp_manager.suggest_vp( gid ) );
+  Node* node = get_node_or_proxy( gid, t );
 
-  Node* node = local_nodes_[ t ].get_node_by_gid( gid );
+  return node;
+}
+
+Node* NodeManager::get_mpi_local_node_or_device_head( index gid )
```

Comment:
It's because if we have a recording device, functions like `get_status` uses the node on thread zero to collect the information from the other nodes. So the node on the first thread acts as a sort of "head" node. When we call `get_mpi_local_node_or_device_head` and we have a recording device, we want this head node.